gtk/pyproject.toml

```diff
@@ -67,7 +67,9 @@ root = ".."
 dependencies = [
     "pycairo >= 1.17.0",
     # PyGObject 3.50.0 is the first version with native GTK asyncio integration.
-    "pygobject >= 3.50.0",
+    # However, from 3.52.1 access to VFuncInfo objects of virtual functions has
+    # been streamlined.
+    "pygobject >= 3.52.1",
```

Comment:
It turns out we can't make this change (at least not yet) - it turns out we [need to allow < 3.52.1 to retain support for Debian 12 and other distros that don't provide libgirepository 2.0](https://github.com/beeware/toga/issues/3862#issuecomment-3515431840).

However, it's not reasonable to stick with an ancient Debian distribution *and* expect to use the most recent GTK version, so rather than restore the older vfunc handling, I'm going to add a `gtk4` optional dependency to make it easy to adopt the dependency, and add error handling for older PyGObject versions on GTK4.